flashrank/Ranker.py

```diff
@@ -210,7 +214,7 @@ def rerank(self, request: RerankRequest) -> List[Dict[str, Any]]:
             raw_ranks = self.llm_model.create_chat_completion(messages)
             results = []
             for rank in raw_ranks["choices"][0]["message"]["content"].split(" > "):
-                results.append(result_map[int(rank[1])])
+                results.append(result_map[self._extract_rank(rank)])
```

Comment:
@nmohr192 - looks like a valid fix.

Since we know the rank will always be enclosed in square brackets `'[]'` _(prepared few lines above)_ we can simplify the code by using `strip("[]")` instead of the regular expression. This will keep it more concise and efficient.

```python
results.append(result_map[int(rank.strip("[]"))])
```

Reply:
I feel like your code is a bit more unintuitive while regex is pretty straightforward and explicit in what is trying to be achieved, as well as the error message when the LLM fails to output properly formatted text. Also the latency difference (efficiency that you mentioned) is negligible and in the nanoseconds.